### ISSUE TITLE
Clean-memory assertion failed

### ISSUE THREAD

**Issue by suntong** [imatix/gsl#21]
Hi again, 

This bug was hard to demonstrate. I've tried to scale down my script as much as possible, but it's still a bit big though. Please check out `memory-bug.gsl` & `memory-bug.xml` from `https://github.com/suntong001/gsl/tree/master/doc/examples`. 

This is how it runs:

```
$ gsl4 -script:memory-bug.gsl memory-bug.xml > /dev/null 
2013/12/26 17:47:41: GSL/4.1a/github Copyright (c) 1996-2011 iMatix Corporation
2013/12/26 17:47:41: gsl/4 I: Processing memory-bug.xml...
Clean-memory assertion failed - <Unknown> (0)
Details are in memtrace.xml
Aborted

$ cat memtrace.xml
<?xml version="1.0"?>
<transaction>
<block index="0" size="16" file="<Unknown>" line="0" pointer="0x10fb7c0" />
</transaction>
```

This is the best I can do to scale down my script. I know that if I remove either the following part, the `Clean-memory assertion failed` problem will be gone:
- the `for` loop in gsl
- the `if` statement after the `for` loop in gsl
- the `Internal getopt error` line from .xml

Please investigate.  
Thanks

**Comment by yamashi** [comment]
Hello,

I am having the same issue, same memtrace.xml file, has anyone figured out a solution for that ?

**Comment by suntong001** [comment]
That's a good news to me because now the status of this bug should be
changed from single-person-reported to confirmed. Hope that will mount the
pressure a bit.

Is it possible for you to provide a scale-down version of your script just
to demo the bug?

Thanks

**Comment by jschultz** [comment]
Hi there,

That's funny I thought I'd fixed this bug.

Suntong - I can't seem to find your memory-bug files that generate the
error, can you help me?

Cheers,
Jonathan

**Comment by freakyjoe** [comment]
Correct location of the files to reproduce: https://github.com/suntong001/gsl_2013-12-26/blob/master/doc/examples/

b/rgds
Joe

**Comment by freakyjoe** [comment]
Just checked: gsl script runs through fine until EOF. Assertion happens on internal cleanup after file is done.

Stripped down code to reproduce:
See: https://gist.github.com/freakyjoe/b0ba400fb9d41f559a21

**Comment by yamashi** [comment]
I integrated your fix and am now getting for every file the error 
(xxx.gsl 1) Unknown function: 
or
  Fatal error in application - aborted
  Abort at GGCODE:Evaluate-Number-Node: (Evaluating, Number-Event)

**Comment by suntong001** [comment]
On Thu, Feb 26, 2015 at 7:38 AM, yamashi wrote:

> I integrated your fix and am now getting for every file the error
> (xxx.gsl 1) Unknown function:
> or
> Fatal error in application - aborted
> Abort at GGCODE:Evaluate-Number-Node: (Evaluating, Number-Event)
>
A stripped down code to reproduce, perhaps?

**Comment by jschultz** [comment]
I believe this issue has been resolved by #72
